lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +678,143 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+                return StatusEvent.
+                        getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                StatusEventType.DELETE_ALL_VERSIONS).
+                        withNullableAccessGroupID(wsid).
+                        withNullableObjectID(Long.toString(objid)).build();
+            }
+        } catch (IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch (JsonClientException ex) {
+            // really think it should be IndexingException here
+            // but IndexingException has a protected constructor
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        }
+
+        // check if name has changed and update state of lastestName
+        ObjectSpecification os = new ObjectSpecification().withRef(
+        ev.getAccessGroupId().get().toString() + "/" +
+                        ev.getAccessGroupObjectId().get());
+
+        final List<ObjectSpecification> getInfoInput = new ArrayList<>();
+        getInfoInput.add(os);
+
+        command = new HashMap<>();
+        command.put("command", "getObjectInfo");
+        command.put("params", new GetObjectInfo3Params().withObjects(getInfoInput));
+
+        final GetObjectInfo3Results objInfo;
+        try {
+            objInfo = ws.getClient().administer(new UObject(command))
+                                        .asClassInstance(GetObjectInfo3Results.class);
+            String name = objInfo.getInfos().get(0).getE2();
+            if (!Utils.isNullOrEmpty(name)) {
+                    latestName = name;
+            }
+        } catch (IOException e) {
+            throw handleException(e);
+        } catch (JsonClientException e) {
+            throw handleException(e);
+        }
+
+        // check if permissions have changed and update state of isPublic
+        try {
+            command = new HashMap<>();
+            command.put("command", "getWorkspaceInfo");
+            command.put("params", new WorkspaceIdentity()
+                            .withId((long) ev.getAccessGroupId().get()));
+
+            final String isPublic;
+
+            isPublic = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(WS_INFO_TYPEREF).getE6();
+
+            if (!Utils.isNullOrEmpty(isPublic)) {
+                latestIsPublic = (isPublic == "n") ? true: false;
+            }
+        } catch (IOException | JsonClientException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        }
+
+        StatusEvent updatedEvent = StatusEvent.
+                getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(), latestEventType).
+                withNullableAccessGroupID(ev.getAccessGroupId().get()).
+                withNullableObjectID(ev.getAccessGroupObjectId().get()).
+                withNullableVersion(ev.getVersion().get()).
```

Comment:
Not all events will have versions - use orNull() instead of get().

Reply:
Done.